### DISCUSSION TITLE
Nextflow 23.10.1-all and custom plugins

### DISCUSSION THREAD

**Discussion by kirrie12** [nextflow-io/nextflow#4748] [Q&A] [answered]
I'm working with the nf-core v.3.14.0 rna-seq pipeline with the 23.10.1 Nextflow`-all` version. I'd like to run the pipeline in an offline environment. I ran Nextflow + pipeline once in an online environment and transferred the files to my offline environment. 

I am aware that the [nf-core documentation](https://nf-co.re/docs/usage/offline) states that the `-all` package does not allow the use of custom plugins. Using the normal Nextflow version results in the attempt to download all the dependencies, which can be avoided by using the `-all` version.

However, user config plugins are ignored in `all`, specifically nf-validation@1.1.3 . 
Pinning the nf-validation@1.1.3 in the configuration files or specifying on the command line does not resolve it. 

How can I make Nextflow and the pipeline work in the offline environment? Thanks in advance!

```
./nextflow-23.10.1-all -c nextflow.config run nf-core-rnaseq/3_14_0 -profile singularity, offline -params-file nf-params.json --max_memory 250.GB
N E X T F L O W  ~  version 23.10.1
Launching `nf-core-rnaseq/3_14_0/main.nf` [gigantic_solvay] DSL2 - revision: 746820de9b
WARN: Nextflow self-contained distribution allows only core plugins -- User config plugins will be ignored: nf-validation@1.1.3
ERROR ~ Cannot find latest version of nf-validation plugin
```

nextflow.config: 
```
plugins{
    id 'nf-validation@1.1.3'
}
```

**Comment by bentsherman** [comment, marked as answer]
You need to find the list of core plugins associated with 23.10.1-edge and download them separately, then use the regular Nextflow install instead of the -all release. As long as the plugins are already downloaded, Nextflow should not try to download them again.

**Reply by kirrie12** [reply]
Thanks for the quick response! I'll try it out and get back to you.

**Reply by kirrie12** [reply]
I managed to resolve the issue, thanks again!
